### PR TITLE
Added Curator's Ward to Dominaria

### PR DESCRIPTION
**Pull request by keelahnkhan**
Added Curator's Ward to Dominaria set https://scryfall.com/card/dom/49

Note: I couldn't find an ability for enchanted target leaving the battlefield or moving zones, hence the custom ability class.

### CODE REVIEW

File: Mage.Sets/src/mage/cards/c/CuratorsWard.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    @Override
+    public CuratorsWard copy() {
+        return new CuratorsWard(this);
+    }
+}
+
+class CuratorsWardTriggeredAbility extends TriggeredAbilityImpl {
+
+    public CuratorsWardTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new DrawCardSourceControllerEffect(2).setText("When enchanted permanent leaves the battlefield, if it was historic, draw two cards."), false);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/c/CuratorsWard.java]: You don't need to do `setText`, instead add `staticText = "When enchanted permanent leaves the battlefield, if it was historic, draw two cards."` to the constructor.

File: Mage.Sets/src/mage/cards/c/CuratorsWard.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.ZONE_CHANGE;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        ZoneChangeEvent zEvent = (ZoneChangeEvent) event;
+        if (zEvent.getFromZone() == Zone.BATTLEFIELD) {
+            Permanent enchanted = game.getPermanentOrLKIBattlefield(event.getTargetId());
+            if (enchanted != null && enchanted.getAttachments().contains(sourceId) && enchanted.isHistoric()) {
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/c/CuratorsWard.java]: Did you test whether this works when both the aura and the enchanted permanent leave at the same time?

File: Mage.Sets/src/mage/cards/c/CuratorsWard.java (hunk 1 above)
Comment (keelahnkhan) [reply Mage.Sets/src/mage/cards/c/CuratorsWard.java]: I tried but it doesn't seem like staticText is an inherited variable of TriggeredAbilityImpl? I couldn't pass the text to the DrawCardSourceControllerEffect constructor either. I might be being oblivious here.

File: Mage.Sets/src/mage/cards/c/CuratorsWard.java (hunk 2 above)
Comment (keelahnkhan) [reply Mage.Sets/src/mage/cards/c/CuratorsWard.java]: I believe so and tested again after you pointed this out to be sure (Akroma's Vengeance was what I tried), the ability still triggers as long as the enchanted permanent is historic. This is how it should work right?

Comment (theelk801) [reply Mage.Sets/src/mage/cards/c/CuratorsWard.java]: Yeah, I just wanted to confirm that based on other similar cards.

File: Mage.Sets/src/mage/cards/c/CuratorsWard.java (hunk 1 above)
Comment (theelk801) [reply Mage.Sets/src/mage/cards/c/CuratorsWard.java]: You're right, instead add
```@Override public String getRule(){return "When enchanted permanent leaves the battlefield, if it was historic, draw two cards.";}

Comment (keelahnkhan) [reply Mage.Sets/src/mage/cards/c/CuratorsWard.java]: Gotcha, will do!

**Pull request merged**